### PR TITLE
Fix Rigidbody copy

### PR DESCRIPTION
**Pull request by CarlCochet**
PR for https://github.com/huggingface/simulate/issues/291

The physics component of an object is now copied with it. 

This PR also adds a custom copy mechanism for `RigidBodyComponent` and `ArticulationBodyComponent`.

### CODE REVIEW

**Comment by CarlCochet** [comment]
@thomwolf

File: src/simulate/assets/articulation_body.py [hunk 1]
```diff
@@ -139,3 +139,35 @@ def __post_init__(self):
...
+
+    def copy(self) -> "ArticulationBodyComponent":
+        """
+        Create a copy of this rigid body with the same values.
+
+        Returns:
+            copy (`RigidBodyComponent`):
+                A copy of this rigid body
+        """
+        instance_copy = ArticulationBodyComponent(
```
Comment (thomwolf) [inline src/simulate/assets/articulation_body.py:151]: We may make these class evolve in the future and we'll miss it here probably.

Let's either use a loop over the fields of the dataclass and/or add a test doing such loop and checking that we didn't forget anything?

File: src/simulate/assets/rigid_body.py [hunk 2]
```diff
@@ -128,3 +128,26 @@ def __post_init__(self):
...
+
+    def copy(self) -> "RigidBodyComponent":
+        """
+        Create a copy of this rigid body with the same values.
+
+        Returns:
+            copy (`RigidBodyComponent`):
+                A copy of this rigid body
+        """
+        instance_copy = RigidBodyComponent(
```
Comment (thomwolf) [inline src/simulate/assets/rigid_body.py:140]: Same here, probably better to loop over the fields and convert/copy/raise an error

**Review by thomwolf — COMMENTED** [review]
Some comments because we may forget to make this evolve in the future

File: src/simulate/assets/articulation_body.py (hunk 1 above)
Comment (CarlCochet) [reply src/simulate/assets/articulation_body.py]: Fixed

File: src/simulate/assets/rigid_body.py (hunk 2 above)
Comment (CarlCochet) [reply src/simulate/assets/rigid_body.py]: Fixed

**Pull request merged**